### PR TITLE
feat: mock download test logic in pluginmanager

### PR DESCRIPTION
**Pull request by steinliber**
Signed-off-by: Meng JiaFeng <jiafeng.meng@merico.dev>

## Pre-Checklist
- [x] I have read through the [CONTRIBUTING.md](https://github.com/devstream-io/devstream/blob/main/CONTRIBUTING.md) documentation.
- [x] My code has the necessary comments and documentation (if needed).
- [x] I have added relevant tests

## Description
mock download logic in unit test

## Related Issues
#827

### CODE REVIEW

File: internal/pkg/pluginmanager/downloader_test.go [hunk 1]
```diff
@@ -1,29 +1,55 @@
...
+}
+
+func mockPlugNotFoundGetter(reqClient *resty.Client, url, plugName string) error {
+	return errors.New("Plug Not Exist")
+}
 
+func TestDownloadSuccess(t *testing.T) {
+	tmpDir := t.TempDir()
+	plugName := "argocdapp_0.0.1-rc1.so"
+	version := "0.0.1-ut-do-not-delete"
```
Comment (algobot76) [inline internal/pkg/pluginmanager/downloader_test.go:26]: Please consider refactoring `plugName` and `version` as constants.

**Review by algobot76 — CHANGES_REQUESTED** [review]

File: internal/pkg/pluginmanager/downloader_test.go [hunk 2]
```diff
@@ -1,29 +1,55 @@
...
 )
 
-func TestDownload(t *testing.T) {
-	os.Remove(filepath.Join(".", "argocdapp_0.0.1-rc1.so"))
+func mockPlugGetter(reqClient *resty.Client, url, plugName string) error {
+	return nil
+}
+
+func mockPlugNotFoundGetter(reqClient *resty.Client, url, plugName string) error {
+	return errors.New("Plug Not Exist")
```
Comment (algobot76) [inline internal/pkg/pluginmanager/downloader_test.go:20]: Please note that when downloading fails, it returns `fmt.Errorf("downloading plugin %s from %s status code %d", plugName, url, response.StatusCode())` NOT `errors.New("Plug Not Exist")`

**Review by algobot76 — CHANGES_REQUESTED** [review]

**Comment by algobot76** [comment]
nitpick: consider refactoring the test cases into a table-driven test

File: internal/pkg/pluginmanager/downloader_test.go (hunk 1 above)
Comment (steinliber) [reply internal/pkg/pluginmanager/downloader_test.go:26]: yeah，constant should be a better choice

File: internal/pkg/pluginmanager/downloader_test.go [hunk 3]
```diff
@@ -1,29 +1,55 @@
...
+	tmpDir := t.TempDir()
+	plugName := "argocdapp_0.0.1-rc1.so"
+	version := "0.0.1-ut-do-not-delete"
 	c := NewDownloadClient()
-	err := c.download(".", "argocdapp_0.0.1-rc1.so", "0.0.1-ut-do-not-delete")
+	tmpFilePath := filepath.Join(tmpDir, fmt.Sprintf("%s.tmp", plugName))
+	_, err := os.Create(tmpFilePath)
 	if err != nil {
-		t.Fatal("downloaded error")
+		log.Fatal("Download logic create tmp file failed")
```
[the remark this answers is not in the document]
Comment (steinliber) [reply internal/pkg/pluginmanager/downloader_test.go:31]: tempDir  will be auto deleted after test finished refer to https://pkg.go.dev/testing#B.TempDir

**Comment by steinliber** [comment]
I changed these tests to be table-driven，It does look better

File: internal/pkg/pluginmanager/downloader_test.go (hunk 2 above)
Comment (IronCore864) [reply internal/pkg/pluginmanager/downloader_test.go]: Titlecase isn't necessary for error messages.

Error messages should not be a full sentence, so the first character should not be uppercase.

**Review by IronCore864 — CHANGES_REQUESTED** [review]
Please resolve all open comments. I think there is one left, which is variables v.s. constants.

**Comment by steinliber** [comment]
I have changed variables to constants, this make code more clear

**Review by IronCore864 — APPROVED** [review]

**Review by hxcGit — APPROVED** [review]

**Pull request merged**